File: Backend/Models/Entities/Temporary_Access.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ProcionAPI.Models.Entities
{
    public class Temporary_Access
    {
        [Key]
        public int Temp_Access_ID { get; set; }

        [Required]
        public int Delegation_ID { get; set; }

        [ForeignKey("Delegation_ID")]
        public Delegation_Of_Authority Delegation_Of_Authority { get; set; }

        [MaxLength(32)]
        public string Name { get; set; }

        [Required]
        public string IsAdmin { get; set; }
        [Required]
        public string CanAccInv { get; set; }
        [Required]
        public string CanAccFin { get; set; }
        [Required]
        public string CanAccPro { get; set; }
        [Required]
        public string CanAccVen { get; set; }
        [Required]
        public string CanAccRep { get; set; }
        [Required]
        public string CanViewPenPro { get; set; }
        [Required]
        public string CanViewFlagPro { get; set; }
        [Required]
        public string CanViewFinPro { get; set; }
        [Required]
        public string CanAppVen { get; set; }
        [Required]
        public string CanEditVen { get; set; }
        [Required]
        public string CanDeleteVen { get; set; }
    }
}

File: IonicFrontend/Procionic/src/app/Shared/Temporary_Access.ts
import { Delegation_Of_Authority } from "./DelegationOfAuthority";

export interface Temporary_Access {
  temp_Access_ID: Number;
  delegation_ID: Number;
  delegation_Of_Authority: Delegation_Of_Authority;
  name: String;
  IsAdmin: string;
  CanAccInv: string;
  CanAccFin: string;
  CanAccPro: string;
  CanAccVen: string;
  CanAccRep: string;
  CanViewPenPro: string;
  CanViewFlagPro: string;
  CanViewFinPro: string;
  CanAppVen: string;
  CanEditVen: string;
  CanDeleteVen: string;
}
